### ISSUE TITLE
"KeyError: Key 32 not found" in solution.jl

### ISSUE THREAD

**Issue by Dx-Wang** [lanl-ansi/PowerModelsMLD.jl#28]
I tried to run the AC MLD on the [ACTIVSg200 system.](https://electricgrids.engr.tamu.edu/electric-grid-test-cases/activsg200/) after setting the `bus_type` of some buses to `4`. The program seems to be running fine until exporting the solution. The complete error message is attached below. The problem seems to be at `C:\Users\username\.julia\packages\PowerModels\w29A2\src\core\solution.jl:249`.

Is there a fix or workaround? Thanks!

```
Number of objective function evaluations             = 47
Number of objective gradient evaluations             = 43
Number of equality constraint evaluations            = 47
Number of inequality constraint evaluations          = 47
Number of equality constraint Jacobian evaluations   = 43
Number of inequality constraint Jacobian evaluations = 43
Number of Lagrangian Hessian evaluations             = 42
Total CPU secs in IPOPT (w/o function evaluations)   =      2.908
Total CPU secs in NLP function evaluations           =      0.739

EXIT: Optimal Solution Found.
[warn | PowerModels]: the given optimizer does not provide the SolveTime() attribute, falling back on @timed.  This is not a rigorous timing value.
ERROR: LoadError: KeyError: key 32 not found
Stacktrace:
 [1] getindex at .\dict.jl:478 [inlined]
 [2] lookup_index at C:\Users\username\.julia\packages\JuMP\ibcEh\src\Containers\DenseAxisArray.jl:136 [inlined]
 [3] _to_index_tuple at C:\Users\username\.julia\packages\JuMP\ibcEh\src\Containers\DenseAxisArray.jl:145 [inlined]
 [4] to_index at C:\Users\username\.julia\packages\JuMP\ibcEh\src\Containers\DenseAxisArray.jl:163 [inlined]
 [5] getindex(::JuMP.Containers.DenseAxisArray{JuMP.VariableRef,1,Tuple{Base.KeySet{Int64,Dict{Int64,Any}}},Tuple{Dict{Int64,Int64}}}, ::Int64) at C:\Users\username\.julia\packages\JuMP\ibcEh\src\Containers\DenseAxisArray.jl:177
 [6] #add_setpoint!#502(::String, ::getfield(PowerModels, Symbol("##507#514")), ::getfield(PowerModels, Symbol("##508#515")), ::getfield(PowerModels, Symbol("##509#516")), ::Dict{String,Any}, ::Bool, ::String, ::Int64, ::typeof(add_setpoint!), ::Dict{String,Any}, ::SOCWRPowerModel, ::String, ::String, ::Symbol) at C:\Users\username\.julia\packages\PowerModels\w29A2\src\core\solution.jl:249
 [7] (::getfield(PowerModels, Symbol("#kw##add_setpoint!")))(::NamedTuple{(:status_name,),Tuple{String}}, ::typeof(add_setpoint!), ::Dict{String,Any}, ::SOCWRPowerModel, ::String, ::String, ::Symbol) at .\none:0
 [8] add_setpoint_generator_power! at C:\Users\username\.julia\packages\PowerModels\w29A2\src\core\solution.jl:117 [inlined]
 [9] solution_mld(::SOCWRPowerModel, ::Dict{String,Any}) at C:\Users\username\.julia\packages\PowerModelsMLD\lLm9c\src\prob\mld.jl:108
 [10] #build_result#469(::typeof(solution_mld), ::Function, ::SOCWRPowerModel, ::Float64) at C:\Users\username\.julia\packages\PowerModels\w29A2\src\core\solution.jl:33
 [11] (::getfield(PowerModels, Symbol("#kw##build_result")))(::NamedTuple{(:solution_builder,),Tuple{typeof(solution_mld)}}, ::typeof(build_result), ::SOCWRPowerModel, ::Float64) at .\none:0
 [12] #optimize_model!#199(::JuMP.OptimizerFactory, ::Function, ::Function, ::SOCWRPowerModel) at C:\Users\username\.julia\packages\PowerModels\w29A2\src\core\base.jl:222
 [13] (::getfield(PowerModels, Symbol("#kw##optimize_model!")))(::NamedTuple{(:optimizer, :solution_builder),Tuple{JuMP.OptimizerFactory,typeof(solution_mld)}}, ::typeof(optimize_model!), ::SOCWRPowerModel) at .\none:0
 [14] #run_model#196(::Array{Any,1}, ::Function, ::Base.Iterators.Pairs{Symbol,Dict{String,Any},Tuple{Symbol},NamedTuple{(:setting,),Tuple{Dict{String,Any}}}}, ::Function, ::Dict{String,Any}, ::Type, ::JuMP.OptimizerFactory, ::Function) at C:\Users\username\.julia\packages\PowerModels\w29A2\src\core\base.jl:151
 [15] (::getfield(PowerModels, Symbol("#kw##run_model")))(::NamedTuple{(:solution_builder, :setting),Tuple{typeof(solution_mld),Dict{String,Any}}}, ::typeof(run_model), ::Dict{String,Any}, ::Type, ::JuMP.OptimizerFactory, ::Function) at .\none:0
 [16] #run_mld#119(::Base.Iterators.Pairs{Symbol,Dict{String,Any},Tuple{Symbol},NamedTuple{(:setting,),Tuple{Dict{String,Any}}}}, ::Function, ::Dict{String,Any}, ::Type, ::JuMP.OptimizerFactory) at C:\Users\username\.julia\packages\PowerModelsMLD\lLm9c\src\prob\mld.jl:3
 [17] (::getfield(PowerModelsMLD, Symbol("#kw##run_mld")))(::NamedTuple{(:setting,),Tuple{Dict{String,Any}}}, ::typeof(run_mld), ::Dict{String,Any}, ::Type, ::JuMP.OptimizerFactory) at .\none:0
 [18] #run_ac_mld_uc#143(::Dict{String,Any}, ::Dict{String,Any}, ::Float64, ::Function, ::Dict{String,Any}, ::JuMP.OptimizerFactory) at C:\Users\username\.julia\packages\PowerModelsMLD\lLm9c\src\util\ac-mld-uc.jl:16
 [19] run_ac_mld_uc(::Dict{String,Any}, ::JuMP.OptimizerFactory) at C:\Users\username\.julia\packages\PowerModelsMLD\lLm9c\src\util\ac-mld-uc.jl:4
 [20] top-level scope at none:0
 [21] include_string(::Module, ::String, ::String) at .\loading.jl:1008
 [22] (::getfield(Main._vscodeserver, Symbol("##9#12")){String,Int64,Int64,String})() at c:\Users\username\.vscode\extensions\julialang.language-julia-0.13.1\scripts\terminalserver\terminalserver.jl:153
 [23] withpath(::getfield(Main._vscodeserver, Symbol("##9#12")){String,Int64,Int64,String}, ::String) at c:\Users\username\.vscode\extensions\julialang.language-julia-0.13.1\scripts\terminalserver\repl.jl:62
 [24] (::getfield(Main._vscodeserver, Symbol("##8#11")){String,Int64,Int64,String})() at c:\Users\username\.vscode\extensions\julialang.language-julia-0.13.1\scripts\terminalserver\terminalserver.jl:152
 [25] hideprompt(::getfield(Main._vscodeserver, Symbol("##8#11")){String,Int64,Int64,String}) at c:\Users\username\.vscode\extensions\julialang.language-julia-0.13.1\scripts\terminalserver\repl.jl:28
 [26] macro expansion at c:\Users\username\.vscode\extensions\julialang.language-julia-0.13.1\scripts\terminalserver\terminalserver.jl:148 [inlined]
 [27] (::getfield(Main._vscodeserver, Symbol("##7#10")))() at .\task.jl:259
in expression starting at c:\Users\username\Code_Data\AGM_TDC_Restoration\tdc-topology\postHurricaneMLD.jl:30
```

**Comment by ccoffrin** [comment]
Could you give it a try with PowerModelsRestoration?  I have moved this functionality over to that one.  If you still have a problem, can you post a small example highlighting the operations that you are using?

**Comment by Dx-Wang** [comment]
Thanks for your suggestion. I tried PowerModelsRestoration. The issue still exists. I've attached a minimal example including .m and .jl files [here](https://github.com/lanl-ansi/PowerModelsMLD.jl/files/4151483/example.zip).

**Comment by noahrhodes** [comment]
You may need to use the function `propagate_topology_status` from PowerModels to change the statuses of the branches connected to buses you set to bus_type 4.

The SOCWR model will try to find the two connected buses of a branch, but if the bus_type is 4 the bus variables will not be created in the optimization model.  Therefore, the branch's status must also be set to inactive.

**Comment by ccoffrin** [comment]
I believe this has been addressed with the release of PowerModel v0.14.2.  @Dx-Wang, can you confirm and if so, close this issue?

Also, this feature should be working with PowerModelsRestoration v0.1.1, so I would recomend using PowerModelsRestoration in place of PowerModelsMLD.

**Comment by Dx-Wang** [comment]
Yes. The issue is resolved with PowerModel v0.14.2 and PowerModelsRestoration v0.1.1. Thanks a lot!
